### ISSUE TITLE
Unable to remove config item and possibly incorrect module name returned from Get-PSFConfig

### ISSUE THREAD

**Issue by dtrue** [PowershellFrameworkCollective/psframework#484]
Hi, I have two issues/questions that I'm trying to resolve that may or may not be related.  When I run Get-PSFConfig I see the list of configuration members by fullname/value etc but the module being returned is incorrect.  Or it was inadvertently loaded at one point with the wrong module name for which now I can't seem to remove.  One tidbit:  the module name has a period in it which may be the cause?  For example, my module is COI.Utility and an example full name would be COI.Utility.SmsNumber.  In this case the module is showing up as COI rather than COI.Utility.  It also seems to revert back and forth (sometimes showing COI.Utility correctly).

I've tried to remove this config item all together using Remove-Config yet it still seems to re-appear although I don't see it in the code base.  The local .json file I see does not show this property so I assume it's in a cache or registry or somewhere else.

So basically two items:
1) Is there an issue with having a module name using a period in it?
2)How to truly remove a config item?

Thanks so much!  Great module by the way!!

**Comment by FriedrichWeinmann** [comment]
heya @dtrue ,
sorry for the trouble :(

Let's deal with it in order then:

> Is there an issue with having a module name using a period in it?

Yeah, I messed up historically with the name processing, wanting to guide people into thinking in namespacing their settings with their modules, to have multiple modules not interfere with each other (For example, imagine two modules both used a setting only named `SmsNumber`, no context).
At the same time, I wanted to make it simple to just access the setting, which birthed the FullName Property.
In the backend they are stored under their FullName, so _technically_ it's an academic difference whether a setting is `Module: COI Name: Utility.SmsNumber` or `Module: COI.Utility Name: SmsNumber` .

That said, the "Module" bit is really an unrelated piece of text that has no effect on your PowerShell module, so you don't have to literally use the same naming scheme as your module - switching to `COIUtility.SmsNumber` or `COI_Utility.SmsNumber` would in no way cause trouble as well.

Sticking with the original scheme is fine too, just make sure to always use the full name of the config item, as depending on how the setting is loaded, the split between Module and Name may be wrong.

> How to truly remove a config item?

The configuration system was originally designed as a kind of options menu for modules. It is by no means constrained to just that and the scenarios I have in mind for it have evolved since then, but ... that's still the roots of it. Deleting settings was never an expected situation - it would only happen in between module versions and you'd just load the new version in a new session, right?
Anyway, if you need to remove it from memory, special measures need be taken, as detailed below.
If however the setting sticks across sessions, you must have accidentally (or not) _persisted_ the setting and need to unregister it.

+ [Deleting Settings](https://psframework.org/documentation/documents/psframework/configuration/deleting-settings.html)
+ [Persistence Basics](https://psframework.org/documentation/documents/psframework/configuration/persistence-basics.html)

**Comment by dtrue** [comment]
Hmm - yeah I can't quite reproduce but it seems as though the Module displayed when running Get-PSFConfig changes back and forth between COI and COI.Utility!  But at the end of the day if thats just a visual quirk I don't really care.  It does make it harder to have users remember they may need to search on module of both values if one doesnt return!
More importantly though - so if a configuration is loaded using the -initialize parameter is it safe that it can NEVER be removed?  Meaning, say I inadvertently loaded one with the -Initialize parameter without a -AllowDelete and I messed up the name.  Can I later delete it or is it always going to be returned in the Get-PSFConfig results?

Thanks so much in advance!
Dan

**Comment by FriedrichWeinmann** [comment]
Yepp Dan, that's exactly it:
The `-Initialize` call is your last opportunity to control that behavior - it is assumed that the `-Initialize` call is kind of the "manifest" declaration of each setting, defining all its metadata.
After that, your only choice is to restart the console

**Comment by dtrue** [comment]
No I mean can I not delete at all?  I don't mind a starting a new console/session.  I'm literally asking is there no way to remove it from my system ever?  Meaning, I don't see reference to this configuration item anywhere in the codebase anymore.  And if I remove-module and load-module it still appears in the Get-PSFConfig.
Sorry if I'm missing something!
Thanks,
Dan

**Comment by FriedrichWeinmann** [comment]
Well, you absolutely can remove it permanently!
But one thing right away: The config system is _processwide_ not a runspace sessionstate thing.
That means Remove-Module and Import-Module has exactly no effect on it.
Complete restart of the console clears any in-memory items.

> but!

Configuration can be _persisted_ by using `Register-PSFConfig` which writes to [one of seven locations](https://psframework.org/documentation/documents/psframework/configuration/persistence-location.html) (which one depends on the `-Scope` parameter). If those exist, they will be loaded each time you import PSFramework.
You can either manually check these locations and clear them or use `Unregister-PSFConfig` to do the job for you

**Comment by dtrue** [comment]
Yeah I guess that's the issue.  Unregister-PSFConfig command does not remove the item in question permanently.  Get-PSFConfig continues to show that item if you open a new pwsh session.
